plugins/web/opentelemetry-instrumentation-user-interaction/src/instrumentation.ts

```diff
@@ -36,6 +36,22 @@ import { VERSION } from './version';
 const ZONE_CONTEXT_KEY = 'OT_ZONE_CONTEXT';
 const EVENT_NAVIGATION_NAME = 'Navigation:';
 
+/**
+ * Most browser provide event listener api via EventTarget in prototype chain.
+ * Exception to this is IE 11 which has it on the prototypes closest to EventTarget:
+ *
+ * * - has addEventListener in IE
+ * ** - has addEventListener in all other browsers
+ * ! - missing in IE
+ *
+ * HTMLElement -> Element -> Node * -> EventTarget **! -> Object
+ * Document -> Node * -> EventTarget **! -> Object
+ * Window * -> WindowProperties ! -> EventTarget **! -> Object
+ */
+const EVENT_TARGETS = window.EventTarget
```

Comment:
As ```isWrapped``` handles detection of whether a function has already been wrapped, why not just always list the entry points that we want to wrap, rather than the split (which really would require tests to be run against IE to ensure this doesn't get broken).

Reply:
This also assumes (which is generally true), that the root object names "Node", "Window" "EventTarget" always exist in all runtimes (which they *may* not).

Would suggest that we have an additional "wrap" function that can take the ```target``` and ```funcName``` so that it can do the following and wrap the correct level.
- Check if ```target``` is not null/undefined (if true -- nothing to wrap => abort wrap)
- Check if ```target[funcName]``` is defined (effectively check for instance level override) (if true wrap instance function (for wrapping simple polyfill's); if false check next)
- Get ```prototype``` of ```target``` (this can be tricky for all environments) 
- Check if ```targetPrototype[funcName]``` exists (if true wrap; otherwise abort wrap)

Examples of getting the correct target are in the ApplicationInsights code I wrote, this is more than what OT needs as it supports a larger set of runtimes (ie. all browsers (IE7+), Node, React, Angular, ReactNative and variants etc.) https://github.com/microsoft/ApplicationInsights-JS/blob/master/shared/AppInsightsCore/src/JavaScriptSDK/InstrumentHooks.ts#L152-L181

Reply:
Just tracked through the ```this._wrap``` which is really the ```shimmer``` project dependency, it does almost all of the above except for the ```prototype``` lookup if it's not defined on the target. It also has a dependency on ```Object.defineProperty``` (which I assume is not the only place in the IT code 😄 ) so my previous comment about IE <= 8 is mute as this just won't work anyway.

Reply:
ping @MSNev was your concerns resolved ?

Reply:
> Think a browser runtime that doesn't support DOM spec essentials like those has bigger issues than opentelemetry support

It's not the browser runtimes you need to be aware of as they generally have them all, it's other native runtimes that "support" JavaScript... 

> maybe this and IE 11 can be thrown away.

I'm still waiting for IE7/8 to be thrown away and as part of the ApplicationInsights work we have to make sure it keeps working (mostly for the health industry) as they tend to lag and still have older versions of Windows and IE. Additionally, when you run IE 11 as an "embedded" browser (create a C# app and drop a browser component on it), by default it runs in IE7 mode.

I haven't checked if you have created separate tests (I suspect not, based on comments and the runtime), so this would mean that this comment is still valid

> why not just always list the entry points that we want to wrap, rather than the split (which really would require tests to be run against IE to ensure this doesn't get broken).

So this really means that we are adding a code path this is never tested (the primary concern)